engine/apps/alerts/models/alert_group.py

```diff
@@ -1599,28 +1594,20 @@ def bulk_silence(user: User, alert_groups: "QuerySet[AlertGroup]", silence_delay
             AlertGroup._bulk_silence(user, root_alert_groups_to_silence, silence_delay)
             AlertGroup._bulk_silence(user, dependent_alert_groups_to_silence, silence_delay)
 
-    def start_ack_reminder(self, user: User):
+    def start_ack_reminder_if_needed(self) -> None:
         from apps.user_management.models import Organization
 
-        unique_unacknowledge_process_id = uuid1()
-        logger.info(
-            f"AlertGroup acknowledged by user with pk "
-            f"{user.pk}, "
-            f"acknowledge timeout task has been started with process id {unique_unacknowledge_process_id}"
-        )
+        if not self.is_root_alert_group:
+            return
 
-        seconds = Organization.ACKNOWLEDGE_REMIND_DELAY[self.channel.organization.acknowledge_remind_timeout]
-        if seconds > 0:
-            delay = datetime.timedelta(seconds=seconds).total_seconds()
-            acknowledge_reminder_task.apply_async(
-                (
-                    self.pk,
-                    unique_unacknowledge_process_id,
-                ),
-                countdown=delay,
-            )
-            self.last_unique_unacknowledge_process_id = unique_unacknowledge_process_id
-            self.save(update_fields=["last_unique_unacknowledge_process_id"])
+        # Check if the "Remind every N hours" setting is enabled
+        countdown = Organization.ACKNOWLEDGE_REMIND_DELAY[self.channel.organization.acknowledge_remind_timeout]
+        if not countdown:
+            return
+
+        self.last_unique_unacknowledge_process_id = celery_uuid()
```

Comment:
The main change of the PR: instead of using `uuid1` to generate Celery task IDs, use `celery.uuid`.

The method stopped working after we upgraded to [kombu 5.3.0](https://github.com/celery/kombu/releases/tag/v5.3.0) (a transitive dependency of celery). Previously, when `UUID` objects were passed to Celery tasks as arguments, they were converted to `str` and it was safe to compare it with strings. After upgrading to kombu 5.3.0, UUIDs are passed to tasks as `UUID` type, thus comparing them with strings always returns `False`.